src/components/stateTracker/js/StateTracker.js

```diff
@@ -0,0 +1,100 @@
+                funcName: "fluid.stateTracker.startTracking",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                                 // changeEvaluator, changeListener, polling interval
+            },
+            stopTracking: {
+                funcName: "fluid.stateTracker.stopTracking",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+                                 // changeListener, intervalID
+            },
+            monitorChange: {
+                funcName: "fluid.stateTracker.monitorChange",
+                args: ["{that}", "{arguments}.0"]
+                                 // monitor info structure
+            },
+            initMonitorInfo: {
+                funcName: "fluid.stateTracker.initMonitorInfo",
+                args: ["{arguments}.0"]
+                       // changeEvaluator object or function
+            }
+        }
+    });
+
+    // Initiate polling.
+    // @param changeEvaluator - an object used to evaluate if the state
+    //                          has changed.  It must have an evaluateChange()
+    //                          function.
+    // @param changeListener  - a listener to handle the onStateChange event.
+    // @param interval        - optional delay between calls to check the state's
+    //                          current value (msec).
+    // @return                - the intervalID.
+    fluid.stateTracker.startTracking = function (that, changeEvaluator, changeListener, interval) {
+        var monitor = that.initMonitorInfo(changeEvaluator);
+        that.events.onStateChange.addListener(changeListener);
+        if (!!interval) {
+            that.interval = interval;
+        }
+        monitor.intervalID = setInterval(function () {
+            that.monitorChange(monitor);
+        }, that.interval);
+        return monitor.intervalID;
+    };
+
+    // Stop polling.
+    // @param changeListener - the listener to remove.
+    // @param intervalID     - the interval to clear.
+    fluid.stateTracker.stopTracking = function (that, listener, intervalID) {
+        that.events.onStateChange.removeListener(listener);
+        clearInterval(intervalID);
+    };
+
+    // Callback to pass to setInterval() to periodically check a state.
+    fluid.stateTracker.monitorChange = function (that, monitor) {
+        if (monitor.changeEvaluator.evaluateChange()) {
+            that.events.onStateChange.fire(monitor.changeEvaluator);
+        }
+    };
+
+    // Initialize a monitor object for passing to a function that periodically
+    // checks for chanages in the state of a process.
+    // @param changeEvaluator - an object to evaluate if the state has changed
+    //                          that has a function named evaluateChange().
+    //                          Provided by client.
+    // @return                - the monitor object.
+    fluid.stateTracker.initMonitorInfo = function (changeEvaluator) {
```

Comment:
I had considered making the evaluator a function even when this code was part of the ProcessReporter.  It was just easier to make it an object that contained whatever other pieces of data or functions the client wanted to stick in there.  The unit tests go so far use a fluid component - see https://github.com/klown/infusion/blob/FLUID-6090/tests/component-tests/stateTracker/js/StateTrackerTests.js#L28

Another way to go is to make the state tracker component a base abstract grade with an evaluateChange invoker defined as `fluid.notImplemented`, as you suggest.  Then, clients would derive a concrete grade and add any data/functionality they want.

Is that what you are suggesting here?  A base abstract grade?